diff -ur ostree.old/src/libostree/ostree-repo-checkout.c ostree/src/libostree/ostree-repo-checkout.c
--- ostree.old/src/libostree/ostree-repo-checkout.c	2021-08-30 19:35:53.000000000 +0200
+++ ostree/src/libostree/ostree-repo-checkout.c	2021-11-03 21:42:58.034678442 +0100
@@ -1389,7 +1389,7 @@
   g_autoptr(GFile) target_dir = NULL;
 
   if (strcmp (options->subpath, "/") != 0)
-    target_dir = g_file_get_child (commit_root, options->subpath);
+    target_dir = g_file_resolve_relative_path (commit_root, options->subpath);
   else
     target_dir = g_object_ref (commit_root);
   g_autoptr(GFileInfo) target_info =
