Index: /Source/WebKit/Platform/IPC/StreamConnectionBuffer.h
===================================================================
--- /Source/WebKit/Platform/IPC/StreamConnectionBuffer.h	(revision 273642)
+++ /Source/WebKit/Platform/IPC/StreamConnectionBuffer.h	(revision 273643)
@@ -115,5 +115,5 @@
     };
     Header& header() const { return *reinterpret_cast<Header*>(m_sharedMemory->data()); }
-    static constexpr size_t headerSize() { return roundUpToMultipleOf<sizeof(std::max_align_t)>(sizeof(Header)); }
+    static constexpr size_t headerSize() { return roundUpToMultipleOf<alignof(std::max_align_t)>(sizeof(Header)); }
 
     size_t m_dataSize { 0 };
