diff --git a/Source/WebCore/platform/graphics/SourceBufferPrivate.h b/Source/WebCore/platform/graphics/SourceBufferPrivate.h
index 189b1ee1a36a..697252b0a64c 100644
--- a/Source/WebCore/platform/graphics/SourceBufferPrivate.h
+++ b/Source/WebCore/platform/graphics/SourceBufferPrivate.h
@@ -45,6 +45,7 @@
 #include <wtf/LoggerHelper.h>
 #include <wtf/Ref.h>
 #include <wtf/text/AtomStringHash.h>
+#include <wtf/UniqueRef.h>
 
 namespace WebCore {
 

