diff -ru WebKit.orig/Source/cmake/OptionsGTK.cmake WebKit/Source/cmake/OptionsGTK.cmake
--- WebKit.orig/Source/cmake/OptionsGTK.cmake	2018-10-02 11:40:57.986556392 +0100
+++ WebKit/Source/cmake/OptionsGTK.cmake	2018-10-02 11:42:45.217474632 +0100
@@ -356,12 +356,6 @@
     set(ENABLE_INTROSPECTION OFF)
 endif ()
 
-# Override the cached variables, gtk-doc and gobject-introspection do not really work when cross-building.
-if (CMAKE_CROSSCOMPILING)
-    set(ENABLE_GTKDOC OFF)
-    set(ENABLE_INTROSPECTION OFF)
-endif ()
-
 # Override the cached variable, gtk-doc does not really work when building on Mac.
 if (APPLE)
     set(ENABLE_GTKDOC OFF)
