diff --git a/elements/components/linux-firmware.bst b/elements/components/linux-firmware.bst
index 0e872f61..d83b3d18 100644
--- a/elements/components/linux-firmware.bst
+++ b/elements/components/linux-firmware.bst
@@ -6,7 +6,7 @@ depends:
 config:
   install-commands:
   - |
-    make DESTDIR="%{install-root}" FIRMWAREDIR="%{install-root}%{prefix}/firmware" install
+    make DESTDIR="%{install-root}" FIRMWAREDIR="%{indep-libdir}/firmware" install
 
 sources:
 - kind: git_tag
