diff --git a/elements/components.bst b/elements/components.bst
index 9e5b15e14..87f5b417e 100644
--- a/elements/components.bst
+++ b/elements/components.bst
@@ -9,9 +9,7 @@ depends:
 - components/asciidoc.bst
 - components/asciidoctor.bst
 - components/aspell.bst
-- components/at-spi2-atk.bst
 - components/at-spi2-core.bst
-- components/atk.bst
 - components/audit.bst
 - components/autoconf-archive.bst
 - components/autoconf.bst
diff --git a/elements/components/at-spi2-atk.bst b/elements/components/at-spi2-atk.bst
index 9d0266fb8..4aee3ce05 100644
--- a/elements/components/at-spi2-atk.bst
+++ b/elements/components/at-spi2-atk.bst
@@ -2,7 +2,6 @@ kind: meson
 
 depends:
 - filename: bootstrap-import.bst
-- filename: components/atk.bst
 - filename: components/at-spi2-core.bst
 - filename: public-stacks/buildsystem-meson.bst
   type: build
diff --git a/elements/components/fuse3.bst b/elements/components/fuse3.bst
index 9b3060bff..2dedbb114 100644
--- a/elements/components/fuse3.bst
+++ b/elements/components/fuse3.bst
@@ -13,6 +13,10 @@ variables:
     -Duseroot=false
 
 public:
+  bst:
+    integration-commands:
+    - |
+      [ -f "%{bindir}/fusermount" ] || ln -s fusermount3 "%{bindir}/fusermount"
   initial-script:
     script: |
       #!/bin/bash
diff --git a/elements/components/gstreamer-plugins-base.bst b/elements/components/gstreamer-plugins-base.bst
index f2c1b206a..f458512b5 100644
--- a/elements/components/gstreamer-plugins-base.bst
+++ b/elements/components/gstreamer-plugins-base.bst
@@ -4,6 +4,7 @@ build-depends:
 - public-stacks/buildsystem-meson.bst
 - components/gobject-introspection.bst
 - components/hotdoc.bst
+#- components/sdl2.bst
 
 depends:
 - bootstrap-import.bst
@@ -20,7 +21,6 @@ depends:
 - components/mesa-headers.bst
 - components/iso-codes.bst
 - components/alsa-lib.bst
-- components/sdl2.bst
 
 variables:
   command-subdir: subprojects/gst-plugins-base
diff --git a/elements/components/gtk3.bst b/elements/components/gtk3.bst
index 1279c03ac..f7f89d91a 100644
--- a/elements/components/gtk3.bst
+++ b/elements/components/gtk3.bst
@@ -8,8 +8,7 @@ build-depends:
 
 depends:
 - bootstrap-import.bst
-- components/atk.bst
-- components/at-spi2-atk.bst
+- components/at-spi2-core.bst
 - components/cairo.bst
 - components/cups.bst
 - components/gdk-pixbuf.bst
diff --git a/elements/platform.bst b/elements/platform.bst
index 884040bb1..9b268221b 100644
--- a/elements/platform.bst
+++ b/elements/platform.bst
@@ -86,9 +86,7 @@ depends:
 - components/alsa-lib.bst
 - components/alsa-plugins.bst
 - components/aspell.bst
-- components/at-spi2-atk.bst
 - components/at-spi2-core.bst
-- components/atk.bst
 - components/cairo.bst
 - components/cups.bst
 - components/libdbus.bst
