diff --git a/meson.build b/meson.build
index f4b629f..8923518 100644
--- a/meson.build
+++ b/meson.build
@@ -71,7 +71,6 @@ endif
 
 common_cflags = cc.get_supported_arguments(test_cflags)
 
-add_project_arguments(cc.get_supported_arguments('-Werror=deprecated-declarations'), language: 'c')
 conf = configuration_data()
 conf.set_quoted('DATADIR', join_paths(get_option('prefix'), get_option('datadir'), 'gupnp-av'))
 conf.set_quoted('VERSION', meson.project_version())
