diff --git a/themes/spinfinity/Makefile.am b/themes/spinfinity/Makefile.am
index b4bc31d..4798eb5 100644
--- a/themes/spinfinity/Makefile.am
+++ b/themes/spinfinity/Makefile.am
@@ -54,4 +54,4 @@ spinfinity.plymouth: $(srcdir)/spinfinity.plymouth.in
 EXTRA_DIST = spinfinity.plymouth.in
 
 install-data-hook:
-	ln -sf /usr/share/pixmaps/system-logo-white.png $(DESTDIR)$(themedir)/header-image.png
+	ln -sf $(logofile) $(DESTDIR)$(themedir)/header-image.png
