diff --git a/configure.ac b/configure.ac
index ada2e4d..5b0f1fa 100644
--- a/configure.ac
+++ b/configure.ac
@@ -17,10 +17,6 @@ AM_MAINTAINER_MODE([enable])
 
 AM_GNU_GETTEXT_VERSION([0.19.8])
 AM_GNU_GETTEXT([external])
-AC_PATH_PROG([INTLTOOL_EXTRACT], [intltool-extract])
-if test x$INTLTOOL_EXTRACT = x; then
-  AC_MSG_ERROR([intltool-extract is required to build plymouth])
-fi
 
 PKG_PROG_PKG_CONFIG
 LT_INIT([dlopen disable-static pic-only])
