diff -ur plymouth.old/themes/bgrt/Makefile.am plymouth/themes/bgrt/Makefile.am
--- plymouth.old/themes/bgrt/Makefile.am	2020-07-09 10:55:09.401661465 +0200
+++ plymouth/themes/bgrt/Makefile.am	2020-07-23 20:06:43.617729724 +0200
@@ -2,12 +2,15 @@
 nodist_theme_DATA = bgrt.plymouth
 
 MAINTAINERCLEANFILES = Makefile.in bgrt.plymouth
-CLEANFILES = bgrt.plymouth
+CLEANFILES = bgrt.plymouth bgrt.plymouth.template
 
-bgrt.plymouth: $(srcdir)/bgrt.plymouth.desktop
+bgrt.plymouth.template: $(srcdir)/bgrt.plymouth.desktop
 	$(AM_V_GEN) \
 	sed -e 's,[@]PLYMOUTH_THEME_PATH[@],$(PLYMOUTH_THEME_PATH),g' \
-	$(srcdir)/bgrt.plymouth.desktop | \
-	    $(MSGFMT) --desktop --template /dev/stdin -d $(top_srcdir)/po -o $@
+	$(srcdir)/bgrt.plymouth.desktop >$@
+
+bgrt.plymouth: bgrt.plymouth.template
+	$(AM_V_GEN) \
+	$(MSGFMT) --desktop --template $< -d $(top_srcdir)/po -o $@
 
 EXTRA_DIST = bgrt.plymouth.desktop
diff -ur plymouth.old/themes/spinner/Makefile.am plymouth/themes/spinner/Makefile.am
--- plymouth.old/themes/spinner/Makefile.am	2020-07-09 10:55:09.416661301 +0200
+++ plymouth/themes/spinner/Makefile.am	2020-07-23 20:06:32.996932965 +0200
@@ -76,13 +76,16 @@
                     throbber-0030.png
 
 MAINTAINERCLEANFILES = Makefile.in spinner.plymouth
-CLEANFILES = spinner.plymouth
+CLEANFILES = spinner.plymouth spinner.plymouth.template
 
-spinner.plymouth: $(srcdir)/spinner.plymouth.desktop
+spinner.plymouth.template: $(srcdir)/spinner.plymouth.desktop
 	$(AM_V_GEN) \
 	sed -e 's,[@]PLYMOUTH_THEME_PATH[@],$(PLYMOUTH_THEME_PATH),g' \
-	$(srcdir)/spinner.plymouth.desktop | \
-	    $(MSGFMT) --desktop --template /dev/stdin -d $(top_srcdir)/po -o $@
+	$(srcdir)/spinner.plymouth.desktop >$@
+
+spinner.plymouth: spinner.plymouth.template
+	$(AM_V_GEN) \
+	$(MSGFMT) --desktop --template $< -d $(top_srcdir)/po -o $@
 
 
 EXTRA_DIST = spinner.plymouth.desktop
