diff -ur eos-repartition.old/dracut/repartition/endless-repartition.sh eos-repartition/dracut/repartition/endless-repartition.sh
--- eos-repartition.old/dracut/repartition/endless-repartition.sh	2020-05-07 16:33:35.337556363 +0200
+++ eos-repartition/dracut/repartition/endless-repartition.sh	2020-05-07 16:35:04.488211539 +0200
@@ -195,6 +195,9 @@
 $preserve_partition"
 fi
 
+# sfdisk does not seem to like to redefinition of sector-size
+parts=$(echo "$parts" | sed -e '/^sector-size:/d')
+
 echo "$parts"
 echo "$parts" | sfdisk --force --no-reread $root_disk
 ret=$?
