diff --git a/docs/test-deps.py b/docs/test-deps.py
index bc7c226bc..f699bf9ec 100755
--- a/docs/test-deps.py
+++ b/docs/test-deps.py
@@ -2,12 +2,13 @@
 # SPDX-License-Identifier: LGPL-2.1+
 
 import sys
-import markdown
+import importlib.metadata
 
 from distutils.version import LooseVersion
 
 # https://github.com/fwupd/fwupd/pull/3337#issuecomment-858947695
-if LooseVersion(markdown.version) < LooseVersion("3.3.3"):
+markdown_version = importlib.metadata.version("markdown")
+if LooseVersion(markdown_version) < LooseVersion("3.3.3"):
     print("python3-markdown version 3.3.3 required for gi-docgen")
     sys.exit(1)
 
