diff --git a/plugins/uefi-capsule/make-images.py b/plugins/uefi-capsule/make-images.py
index 206f3d474..3eb160cf3 100755
--- a/plugins/uefi-capsule/make-images.py
+++ b/plugins/uefi-capsule/make-images.py
@@ -146,8 +146,7 @@ def main(args) -> int:
                 items = Pango.itemize(pctx, label_translated, 0, length, attrs, None)
                 if not items:
                     continue
-                gs = Pango.GlyphString()
-                Pango.shape(label_translated, length, items[0].analysis, gs)
+                gs = Pango.shape(label_translated, length, items[0].analysis)
                 del img, cctx, pctx, layout
 
                 def find_size(fs, f, data):
