From 75690ec8f0d2d3f4b3c5a5e8f24cbe29cba227a9 Mon Sep 17 00:00:00 2001
From: Abderrahim Kitouni <akitouni@gnome.org>
Date: Wed, 19 Jan 2022 12:22:39 +0100
Subject: [PATCH 1/2] components/wayland.bst: build using meson

---
 elements/components/wayland.bst | 19 ++++++++++---------
 1 file changed, 10 insertions(+), 9 deletions(-)

diff --git a/elements/components/wayland.bst b/elements/components/wayland.bst
index bd7810bc9..e95c2b851 100644
--- a/elements/components/wayland.bst
+++ b/elements/components/wayland.bst
@@ -1,16 +1,17 @@
-kind: autotools
+kind: meson
+
+build-depends:
+- public-stacks/buildsystem-meson.bst
 
 depends:
-- filename: bootstrap-import.bst
-- filename: public-stacks/buildsystem-autotools.bst
-  type: build
-- filename: components/expat.bst
-- filename: components/libxml2.bst
-- filename: components/libffi.bst
+- bootstrap-import.bst
+- components/expat.bst
+- components/libxml2.bst
+- components/libffi.bst
 
 variables:
-  conf-local: |
-    --disable-documentation
+  meson-local: >-
+    -Ddocumentation=false
 
 public:
   bst:
-- 
2.34.1

