diff -ru argyll.orig/Jamtop argyll/Jamtop
--- argyll.orig/Jamtop	2018-05-01 11:04:46.820379112 +0100
+++ argyll/Jamtop	2018-05-01 11:02:36.852895586 +0100
@@ -161,6 +161,8 @@
 			 || [ GLOB /usr/lib64 : lib$(lcase).so ] || [ GLOB /usr/lib64 : lib$(lcase).a ]
 			 || [ GLOB /usr/lib/x86_64-linux-gnu : lib$(lcase).so ]
 			 || [ GLOB /usr/lib/x86_64-linux-gnu : lib$(lcase).a ]
+			 || [ GLOB /usr/lib/aarch64-linux-gnu : lib$(lcase).so ]
+			 || [ GLOB /usr/lib/aarch64-linux-gnu : lib$(lcase).a ]
 	         || [ GLOB /usr/lib/i386-linux-gnu : lib$(lcase).so ]
 	         || [ GLOB /usr/lib/i386-linux-gnu : lib$(lcase).a ] {
 				echo "Using system $(UCASE) library" ;
