From 021f97d9be50a857f0cfa1859fc77785d2b7e1f7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Alberts=20Muktup=C4=81vels?= <alberts.muktupavels@gmail.com>
Date: Fri, 31 Oct 2014 23:09:10 +0200
Subject: [PATCH 1/2] crossfade: remove gdkx/x dependency

---
 libgnome-desktop/gnome-bg-crossfade.c | 29 +--------------------------
 1 file changed, 1 insertion(+), 28 deletions(-)

diff --git a/libgnome-desktop/gnome-bg-crossfade.c b/libgnome-desktop/gnome-bg-crossfade.c
index 77098423..f254b313 100644
--- a/libgnome-desktop/gnome-bg-crossfade.c
+++ b/libgnome-desktop/gnome-bg-crossfade.c
@@ -27,13 +27,9 @@
 #include <gio/gio.h>
 
 #include <gdk/gdk.h>
-#include <gdk/gdkx.h>
-#include <X11/Xlib.h>
-#include <X11/Xatom.h>
 #include <gtk/gtk.h>
 
 #include <cairo.h>
-#include <cairo-xlib.h>
 
 #define GNOME_DESKTOP_USE_UNSTABLE_API
 #include "gnome-bg.h"
@@ -365,34 +361,11 @@ animations_are_disabled (GnomeBGCrossfade *fade)
 	return !are_enabled;
 }
 
-static void
-send_root_property_change_notification (GnomeBGCrossfade *fade)
-{
-        long zero_length_pixmap;
-
-        /* We do a zero length append to force a change notification,
-         * without changing the value */
-        XChangeProperty (GDK_WINDOW_XDISPLAY (fade->priv->window),
-                         GDK_WINDOW_XID (fade->priv->window),
-                         gdk_x11_get_xatom_by_name ("_XROOTPMAP_ID"),
-                         XA_PIXMAP, 32, PropModeAppend,
-                         (guchar *) &zero_length_pixmap, 0);
-}
-
 static void
 draw_background (GnomeBGCrossfade *fade)
 {
 	if (gdk_window_get_window_type (fade->priv->window) == GDK_WINDOW_ROOT) {
-                XClearArea (GDK_WINDOW_XDISPLAY (fade->priv->window),
-                            GDK_WINDOW_XID (fade->priv->window),
-                            0, 0,
-                            gdk_window_get_width (fade->priv->window),
-                            gdk_window_get_height (fade->priv->window),
-                            False);
-
-                send_root_property_change_notification (fade);
-
-		gdk_flush ();
+		g_warning ("Crossfade is not supported on ROOT window!");
 	} else {
 		gdk_window_invalidate_rect (fade->priv->window, NULL, FALSE);
 		gdk_window_process_updates (fade->priv->window, FALSE);
-- 
2.23.0

