diff --git a/disas/arm-a64.cc b/disas/arm-a64.cc
index 9fa779e1..a1402a2e 100644
--- a/disas/arm-a64.cc
+++ b/disas/arm-a64.cc
@@ -17,10 +17,8 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-extern "C" {
 #include "qemu/osdep.h"
 #include "disas/dis-asm.h"
-}
 
 #include "vixl/a64/disasm-a64.h"
 
diff --git a/include/disas/dis-asm.h b/include/disas/dis-asm.h
index 2164762b..b89acf7a 100644
--- a/include/disas/dis-asm.h
+++ b/include/disas/dis-asm.h
@@ -9,6 +9,10 @@
 #ifndef DISAS_DIS_ASM_H
 #define DISAS_DIS_ASM_H
 
+#ifdef  __cplusplus
+extern "C" {
+#endif
+
 typedef void *PTR;
 typedef uint64_t bfd_vma;
 typedef int64_t bfd_signed_vma;
@@ -507,4 +511,8 @@ static inline bfd_vma bfd_getb16(const bfd_byte *addr)
 
 typedef bool bfd_boolean;
 
+#ifdef  __cplusplus
+}
+#endif
+
 #endif /* DISAS_DIS_ASM_H */
