diff --git a/gdk/wayland/gdkglcontext-wayland.c b/gdk/wayland/gdkglcontext-wayland.c
index b0ae79a0b7..b27aeced76 100644
--- a/gdk/wayland/gdkglcontext-wayland.c
+++ b/gdk/wayland/gdkglcontext-wayland.c
@@ -80,6 +80,23 @@ gdk_wayland_gl_context_end_frame (GdkDrawContext *draw_context,
   gdk_wayland_surface_notify_committed (surface);
 }
 
+static gboolean
+gdk_wayland_gl_context_make_current (GdkGLContext *context,
+                                     gboolean      surfaceless)
+{
+  GdkDisplay *display = gdk_gl_context_get_display (context);
+  EGLDisplay egl_display = gdk_display_get_egl_display (display);
+
+  if (!GDK_GL_CONTEXT_CLASS (gdk_wayland_gl_context_parent_class)->make_current (context, surfaceless))
+    return FALSE;
+
+  if (surfaceless)
+    return TRUE;
+
+  eglSwapInterval (egl_display, 0);
+  return TRUE;
+}
+
 static void
 gdk_wayland_gl_context_class_init (GdkWaylandGLContextClass *klass)
 {
@@ -90,6 +107,7 @@ gdk_wayland_gl_context_class_init (GdkWaylandGLContextClass *klass)
   draw_context_class->end_frame = gdk_wayland_gl_context_end_frame;
 
   context_class->backend_type = GDK_GL_EGL;
+  context_class->make_current = gdk_wayland_gl_context_make_current;
 }
 
 static void
