From 2cb92a012e900c28cbebe1489f08e9ef56319806 Mon Sep 17 00:00:00 2001
From: Abderrahim Kitouni <abderrahim.kitouni@codethink.co.uk>
Date: Sat, 13 Jul 2024 09:10:10 +0100
Subject: [PATCH 4/4] sysupdate: fix size_t specifier

---
 src/sysupdate/sysupdate.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/sysupdate/sysupdate.c b/src/sysupdate/sysupdate.c
index 870aa04d7c..fcf8c10080 100644
--- a/src/sysupdate/sysupdate.c
+++ b/src/sysupdate/sysupdate.c
@@ -795,7 +795,7 @@ static int context_on_acquire_progress(const Transfer *t, const Instance *inst,
         overall = (unsigned) ((base + scaled) / 100);
         assert(overall <= 100);
 
-        log_debug("Transfer %" PRIu64 "/%zu is %u%% complete (%u%% overall).", i+1, n, percentage, overall);
+        log_debug("Transfer %zu/%zu is %u%% complete (%u%% overall).", i+1, n, percentage, overall);
         return sd_notifyf(/* unset= */ false, "X_SYSUPDATE_PROGRESS=%u\n"
                                               "X_SYSUPDATE_TRANSFERS_LEFT=%zu\n"
                                               "X_SYSUPDATE_TRANSFERS_DONE=%zu\n"
-- 
2.39.2

