diff --git a/src/sysupdate/sysupdated.c b/src/sysupdate/sysupdated.c
index f387494b58..9cf384a431 100644
--- a/src/sysupdate/sysupdated.c
+++ b/src/sysupdate/sysupdated.c
@@ -1333,10 +1333,12 @@ static int target_method_get_version(sd_bus_message *msg, void *userdata, sd_bus
                 return log_error_errno(r, "Failed to run 'systemd-sysupdate list': %m");
 
         version_json = sd_json_variant_by_key(v, "current");
-        if (!version_json)
+        if (!version_json) {
                 version_json = sd_json_variant_by_key(v, "current+pending");
-        if (!version_json)
+                if (version_json)
+                        return sd_bus_reply_method_return(msg, "s", "");
                 return log_sysupdate_bad_json(SYNTHETIC_ERRNO(EPROTO), "list", "Missing key 'current' or 'current+pending'");
+        }
 
         if (sd_json_variant_is_null(version_json))
                 return sd_bus_reply_method_return(msg, "s", "");
