From 91851cba4e2c3aa5a915b4479cb870bc21629d36 Mon Sep 17 00:00:00 2001
From: Valentin David <me@valentindavid.com>
Date: Sun, 24 Aug 2025 23:39:36 +0100
Subject: [PATCH 01/11] linux: enable OnePlus 6's panel

---
 files/linux/fdsdk-config.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/files/linux/fdsdk-config.sh b/files/linux/fdsdk-config.sh
index bd4395419..bcd6d9b54 100644
--- a/files/linux/fdsdk-config.sh
+++ b/files/linux/fdsdk-config.sh
@@ -713,6 +713,7 @@ case "$arch" in
     aarch64)
         module DRM_PANFROST
         module DRM_MSM
+        module DRM_PANEL_SAMSUNG_SOFEF00
     ;;
 esac
 
-- 
2.53.0

