From 3b996a2e18374c4c55df44c63fe0804eac539dbc Mon Sep 17 00:00:00 2001
From: Jordan Petridis <jpetridis@gnome.org>
Date: Wed, 26 Feb 2025 12:00:00 +0200
Subject: [PATCH] components/wayland-protocols: Update to 1.41.0

---
 elements/components/wayland-protocols.bst | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/elements/components/wayland-protocols.bst b/elements/components/wayland-protocols.bst
index bd7172eb93..04db70455a 100644
--- a/elements/components/wayland-protocols.bst
+++ b/elements/components/wayland-protocols.bst
@@ -8,11 +8,11 @@ depends:
 - components/wayland.bst
 
 variables:
   meson-local: >-
     -Dtests=false
 
 sources:
 - kind: git_repo
   url: freedesktop:wayland/wayland-protocols.git
   track: '*.*'
-  ref: 1.40-0-gc7b582cb71ec145026a547dcd8abb9dfeba40541
+  ref: 1.41-0-g71da8bd7f9fd3ea7c5d87ad19ca087eeca137673
-- 
2.48.1

