From e888616981f6e6036e97fb483c6a06d715c05184 Mon Sep 17 00:00:00 2001
From: bbhtt <bbhtt.zn0i8@slmail.me>
Date: Mon, 23 Sep 2024 12:58:16 +0530
Subject: [PATCH 1/2] libjxl: Update to v0.11.0

---
 elements/components/libjxl.bst | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/elements/components/libjxl.bst b/elements/components/libjxl.bst
index 0a6415f6b5..24f6654eff 100644
--- a/elements/components/libjxl.bst
+++ b/elements/components/libjxl.bst
@@ -47,16 +47,16 @@ variables:
     -DJPEGXL_FORCE_SYSTEM_LCMS2=ON
     -DJPEGXL_STATIC=OFF
     -DJPEGXL_TEST_TOOLS=OFF
     -DJPEGXL_WARNINGS_AS_ERRORS=OFF
     -DJPEGXL_BUNDLE_LIBPNG=OFF
     -DJPEGXL_ENABLE_DOXYGEN=ON
 
 sources:
 - kind: git_repo
   url: github:libjxl/libjxl.git
-  track: v0.10.*
+  track: v0.11.*
   exclude:
   - "*rc*"
   - "*snapshot*"
   - "*base*"
-  ref: v0.10.4-0-g91e935dad222b22e7e12eb452f62fe0be3e70b62
+  ref: v0.11.0-0-g4df1e9eccdf86b8df4c0c7c08f529263906f9c4f
-- 
2.48.1

