From a650e3d36d09974e664af9d7796b315478de15f5 Mon Sep 17 00:00:00 2001
From: Valentin David <me@valentindavid.com>
Date: Sun, 24 Aug 2025 23:39:36 +0100
Subject: [PATCH 03/14] linux: enable OnePlus 6's panel

---
 files/linux/fdsdk-config.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/files/linux/fdsdk-config.sh b/files/linux/fdsdk-config.sh
index b8c021d71..5f7be5fb7 100644
--- a/files/linux/fdsdk-config.sh
+++ b/files/linux/fdsdk-config.sh
@@ -715,20 +715,21 @@ enable DRM_AMDGPU_USERPTR
 enable DRM_AMD_ACP
 case "$arch" in
     x86_64)
         remove DRM_I915; module DRM_I915
         module DRM_GMA500
         module DRM_XE
     ;;
     aarch64)
         module DRM_PANFROST
         module DRM_MSM
+        module DRM_PANEL_SAMSUNG_SOFEF00
     ;;
 esac
 
 # NPUs
 enable DRM_ACCEL
 module DRM_ACCEL_QAIC
 case "$arch" in
     x86_64)
         module DRM_ACCEL_IVPU
         module DRM_ACCEL_AMDXDNA
-- 
2.54.0

