From 87677f228802a7ae9ab59d0be403b54b40ed8ce3 Mon Sep 17 00:00:00 2001
From: Valentin David <me@valentindavid.com>
Date: Sat, 25 Oct 2025 15:11:09 +0200
Subject: [PATCH 5/5] components/linux-firmware.bst: Update to 2025102

---
 elements/components/linux-firmware.bst | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/elements/components/linux-firmware.bst b/elements/components/linux-firmware.bst
index 8ac3d9b82..af7a7c57d 100644
--- a/elements/components/linux-firmware.bst
+++ b/elements/components/linux-firmware.bst
@@ -26,4 +26,4 @@ sources:
 - kind: git_repo
   url: kernel:linux/kernel/git/firmware/linux-firmware.git
   track: '[0-9]*'
-  ref: 20251011-0-g49fafa182b23577825976524a3aef8e1bc8878bf
+  ref: 20251021-0-g6a0152a5d1a9d7699e13ca7ed81df2caa1262a77
-- 
2.51.0

