diff --git a/elements/components/icu.bst b/elements/components/icu.bst
index 3beae1a83..d114ac83b 100644
--- a/elements/components/icu.bst
+++ b/elements/components/icu.bst
@@ -36,8 +36,8 @@ public:
 # See https://github.com/sdispater/tomlkit/issues/322
 sources:
 - kind: tar
-  url: github_files:unicode-org/icu/releases/download/release-77-1/icu4c-77_1-src.tgz
-  ref: 588e431f77327c39031ffbb8843c0e3bc122c211374485fa87dc5f3faff24061
+  url: github_files:unicode-org/icu/releases/download/release-78.3/icu4c-78.3-sources.tgz
   version-guess-pattern: '(\d+)_(\d+)'
+  ref: 3a2e7a47604ba702f345878308e6fefeca612ee895cf4a5f222e7955fabfe0c0
 - kind: patch
   path: patches/icu/icu-fix-shared-linking.patch
