diff --git a/elements/components/dbus-base.bst b/elements/components/dbus-base.bst
index ec8241f5f..0084c8b09 100644
--- a/elements/components/dbus-base.bst
+++ b/elements/components/dbus-base.bst
@@ -24,7 +24,6 @@ variables:
     -Dkqueue=disabled
     -Dlaunchd=disabled
     -Dx11_autolaunch=disabled
-    -Dtraditional_activation=false
     -Ddoxygen_docs=enabled
     -Dxml_docs=enabled
 
@@ -34,6 +33,11 @@ config:
     - |
       rm -rf %{install-root}%{localstatedir}/run
 
+# TODO: replace with --disable-traditional-activation
+# https://gitlab.freedesktop.org/dbus/dbus/commit/94b5b236aa0743bacb2139c62d5a6e0143845634
+    - |
+      rm %{install-root}%{libexecdir}/dbus-daemon-launch-helper
+
 public:
   bst:
     split-rules:
