diff --git a/elements/vm/mesa-default.bst b/elements/vm/mesa-default.bst
index 73499eb98..8fc8a30ac 100644
--- a/elements/vm/mesa-default.bst
+++ b/elements/vm/mesa-default.bst
@@ -14,7 +14,8 @@ config:
   install-commands:
   - |
     for i in vulkan/icd.d glvnd/egl_vendor.d OpenCL/vendors lib/dri \
-             lib/d3d vulkan/explicit_layer.d vulkan/implicit_layer.d; do
+             lib/d3d vulkan/explicit_layer.d vulkan/implicit_layer.d \
+             lib/gbm; do
       mkdir -p "%{install-root}%{libdir}/GL/$(dirname "${i}")"
       ln -rs "%{install-root}%{libdir}/GL/default/${i}" "%{install-root}%{libdir}/GL/${i}"
     done
