diff --git a/files/linux/fdsdk-config.sh b/files/linux/fdsdk-config.sh
index 0c37502e3..d696993d2 100644
--- a/files/linux/fdsdk-config.sh
+++ b/files/linux/fdsdk-config.sh
@@ -625,6 +625,7 @@ case "$arch" in
     x86_64|i686)
         remove DRM_I915; module DRM_I915
         module DRM_GMA500
+        module DRM_XE
     ;;
     aarch64)
         module DRM_PANFROST
@@ -1164,6 +1165,7 @@ case "$arch" in
         enable IRQ_REMAP
         enable X86_AMD_PLATFORM_DEVICE
         enable CONFIG_X86_USER_SHADOW_STACK
+        enable PINCTRL_INTEL_PLATFORM
     ;;
 esac
 
