From 15d1f82292ea900ea1d3c206743ec9818d58f0c2 Mon Sep 17 00:00:00 2001
From: Jordan Petridis <jordan@centricular.com>
Date: Mon, 4 Dec 2023 21:20:52 +0200
Subject: [PATCH 01/21] components/linux.bst: Enable DUMMY network driver

---
 elements/components/linux.bst | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/elements/components/linux.bst b/elements/components/linux.bst
index 231e1330d..bbc6868a7 100644
--- a/elements/components/linux.bst
+++ b/elements/components/linux.bst
@@ -340,6 +340,9 @@ config:
       ;;
     esac
 
+    # Dummy network driver
+    module DUMMY
+
     # For wireless networks
     enable WIRELESS
     module CFG80211
-- 
2.43.0


From 30368c0e4ba2c9663073d7a8ceee10d539b592cc Mon Sep 17 00:00:00 2001
From: Jordan Petridis <jordan@centricular.com>
Date: Thu, 7 Dec 2023 20:27:56 +0200
Subject: [PATCH 02/21] components/linux.bst: Enable wireguard

---
 elements/components/linux.bst | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/elements/components/linux.bst b/elements/components/linux.bst
index bbc6868a7..87c04d01d 100644
--- a/elements/components/linux.bst
+++ b/elements/components/linux.bst
@@ -340,6 +340,9 @@ config:
       ;;
     esac
 
+    # Wireguard
+    module WIREGUARD
+
     # Dummy network driver
     module DUMMY
 
-- 
2.43.0


From f117454d99ca9062eda6b9094be2fa63016d2b60 Mon Sep 17 00:00:00 2001
From: Jordan Petridis <jordan@centricular.com>
Date: Thu, 7 Dec 2023 20:41:51 +0200
Subject: [PATCH 03/21] components/linux.bst: Enable more bluetooth drivers and
 devices

and slihtly sort the list
---
 elements/components/linux.bst | 34 +++++++++++++++++++++++++++-------
 1 file changed, 27 insertions(+), 7 deletions(-)

diff --git a/elements/components/linux.bst b/elements/components/linux.bst
index 87c04d01d..c69fcc1ad 100644
--- a/elements/components/linux.bst
+++ b/elements/components/linux.bst
@@ -784,23 +784,25 @@ config:
     enable SERIAL_DEV_CTRL_TTYPORT
 
     # Bluetooth
-    enable BT
-    module BT_HCIBTUSB
-    module BT_HIDP
-    module UHID
+    module BT
+    enable BT_BREDR
+    enable BT_HS
+    enable BT_LE
+
     module BT_ATH3K
+    module BT_BNEP
     module BT_HCIBCM203X
     module BT_HCIBFUSB
+    module BT_HCIBTUSB
+    module BT_HIDP
+    module BT_RFCOMM
     if has PCMCIA; then
       module BT_HCIBLUECARD
       module BT_HCIBT3C
       module BT_HCIDTL1
     fi
-    module BT_HCIBPA10X
-    module BT_HCIBTSDIO
     enable BT_HCIBTUSB_AUTOSUSPEND
     enable BT_HCIBTUSB_MTK
-    module BT_HCIUART
     enable BT_HCIUART_3WIRE
     enable BT_HCIUART_AG6XX
     enable BT_HCIUART_ATH3K
@@ -813,9 +815,25 @@ config:
     enable BT_HCIUART_NOKIA
     enable BT_HCIUART_QCA
     enable BT_HCIUART_RTL
+    enable BT_LE_L2CAP_ECRED
+    enable BT_LEDS
+    enable BT_MSFTEXT
+    enable BT_RFCOMM_TTY
+    module BT_6LOWPAN
+    enable BT_AOSPEXT
+    enable BT_BNEP_MC_FILTER
+    enable BT_BNEP_PROTO_FILTER
+    module BT_HCIBCM4377
+    module BT_HCIBPA10X
+    module BT_HCIBTSDIO
+    module BT_HCIUART
     module BT_HCIVHCI
     module BT_MRVL
     module BT_MRVL_SDIO
+    module BT_MTKSDIO
+    module BT_MTKUART
+    module BT_NXPUART
+    module BT_VIRTIO
 
     # MIDI
     module SND_SEQUENCER
@@ -1022,6 +1040,7 @@ config:
       module I2C_HID_ACPI
     fi
     enable HIDRAW
+    module UHID
 
     # CPU support
     enable PINCTRL
@@ -1629,6 +1648,7 @@ config:
     enable NETFILTER_ADVANCED
     module VETH
     module BLK_DEV_NBD
+    module 6LOWPAN
 
     # Those two modules are required by multipass
     module NETFILTER_XT_MATCH_COMMENT
-- 
2.43.0


From 43d046843aecbd5245ff28a1a36086f52ed21881 Mon Sep 17 00:00:00 2001
From: Jordan Petridis <jordan@centricular.com>
Date: Fri, 8 Dec 2023 11:56:42 +0200
Subject: [PATCH 04/21] components/linux.bst: Enable phy and LED controllers

---
 elements/components/linux.bst | 60 +++++++++++++++++++++++++++--------
 1 file changed, 47 insertions(+), 13 deletions(-)

diff --git a/elements/components/linux.bst b/elements/components/linux.bst
index c69fcc1ad..046f9a426 100644
--- a/elements/components/linux.bst
+++ b/elements/components/linux.bst
@@ -750,6 +750,48 @@ config:
       module VIDEO_IPU3_CIO2
     fi
 
+    # PHY controllers
+    enable GENERIC_PHY
+    enable NETWORK_PHY_TIMESTAMPING
+    module PHY_CAN_TRANSCEIVER
+    module ADIN_PHY
+    module AMD_PHY
+    module AQUANTIA_PHY
+    module AT803X_PHY
+    module BCM54140_PHY
+    module BCM7XXX_PHY
+    module BCM87XX_PHY
+    module BROADCOM_PHY
+    module CICADA_PHY
+    module CORTINA_PHY
+    module DAVICOM_PHY
+    module DP83822_PHY
+    module DP83848_PHY
+    module DP83869_PHY
+    module ICPLUS_PHY
+    module INTEL_XWAY_PHY
+    module LSI_ET1011C_PHY
+    module LXT_PHY
+    module MARVELL_10G_PHY
+    module MARVELL_88Q2XXX_PHY
+    module MARVELL_88X2222_PHY
+    module MARVELL_PHY
+    module MAXLINEAR_GPHY
+    module MEDIATEK_GE_PHY
+    module MICREL_PHY
+    module MICROCHIP_T1S_PHY
+    module MICROSEMI_PHY
+    module MOTORCOMM_PHY
+    module NATIONAL_PHY
+    module NCN26000_PHY
+    module NXP_C45_TJA11XX_PHY
+    module NXP_CBTX_PHY
+    module PHYLIB
+    module QSEMI_PHY
+    module SFP
+    module STE10XP
+    module TERANETICS_PHY
+
     # NVME disks
     module BLK_DEV_NVME
     enable NVME_HWMON
@@ -1739,24 +1781,16 @@ config:
     esac
 
     # LEDS
+    enable LED_TRIGGER_PHY
+    enable LEDS_BRIGHTNESS_HW_CHANGED
+    enable LEDS_TRIGGERS
+    enable NEW_LEDS
     module LEDS_CLASS
+    module LEDS_CLASS_FLASH
     module LEDS_CLASS_MULTICOLOR
-    enable LEDS_BRIGHTNESS_HW_CHANGED
     module LEDS_LP3944
     module LEDS_LT3593
     module LEDS_REGULATOR
-    module LEDS_TRIGGER_ACTIVITY
-    module LEDS_TRIGGER_BACKLIGHT
-    module LEDS_TRIGGER_CAMERA
-    module LEDS_TRIGGER_DEFAULT_ON
-    enable LEDS_TRIGGER_DISK
-    module LEDS_TRIGGER_HEARTBEAT
-    module LEDS_TRIGGER_NETDEV
-    module LEDS_TRIGGER_ONESHOT
-    enable LEDS_TRIGGER_PANIC
-    module LEDS_TRIGGER_PATTERN
-    module LEDS_TRIGGER_TIMER
-    module LEDS_TRIGGER_TRANSIENT
     case "%{arch}" in
       x86_64|i686)
         module LEDS_APU
-- 
2.43.0


From f04db0fafbe1c70e128af73311b5b7d98f509ab2 Mon Sep 17 00:00:00 2001
From: Jordan Petridis <jordan@centricular.com>
Date: Fri, 8 Dec 2023 12:14:25 +0200
Subject: [PATCH 05/21] components/linux.bst: Enable more nvme drivers

---
 elements/components/linux.bst | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/elements/components/linux.bst b/elements/components/linux.bst
index 046f9a426..a865d0e00 100644
--- a/elements/components/linux.bst
+++ b/elements/components/linux.bst
@@ -793,8 +793,10 @@ config:
     module TERANETICS_PHY
 
     # NVME disks
-    module BLK_DEV_NVME
     enable NVME_HWMON
+    module BLK_DEV_NVME
+    module NVME_FC
+    module NVME_TCP
 
     # Memory card readers
     module MISC_RTSX_PCI
-- 
2.43.0


From f70dff29937caeb766f00ad86a5ff2e25f40d8ee Mon Sep 17 00:00:00 2001
From: Jordan Petridis <jordan@centricular.com>
Date: Fri, 8 Dec 2023 12:17:11 +0200
Subject: [PATCH 06/21] components/linux.bst: Enable USB4 and NVMEM support

---
 elements/components/linux.bst | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/elements/components/linux.bst b/elements/components/linux.bst
index a865d0e00..a9e7899f4 100644
--- a/elements/components/linux.bst
+++ b/elements/components/linux.bst
@@ -674,9 +674,14 @@ config:
     module FIREWIRE_SBP2
 
     # USB
+    enable USB
+    enable USB_PHY
     module USB_ROLE_SWITCH
+    module USB4
+    module USB4_NET
     case "%{arch}" in
       x86_64|i686)
+        module USB_LGM_PHY
         module USB_ROLES_INTEL_XHCI
       ;;
     esac
@@ -1097,6 +1102,14 @@ config:
         ;;
     esac
 
+    # NVMEM support
+    # Needed for USB4 and Thunderbolt
+    enable MTD
+    enable NVMEM
+    module NVMEM_LAYOUT_ONIE_TLV
+    module NVMEM_LAYOUT_SL28_VPD
+    module NVMEM_RMEM
+
     # I2C/SMBus
     case "%{arch}" in
       x86_64|i686)
-- 
2.43.0


From 6a852a4de7ad3e26683f9f32e858ea8e5bea06d0 Mon Sep 17 00:00:00 2001
From: Jordan Petridis <jordan@centricular.com>
Date: Fri, 8 Dec 2023 13:29:49 +0200
Subject: [PATCH 07/21] components/linux.bst: Enable hwrandom for physical cpus

---
 elements/components/linux.bst | 15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/elements/components/linux.bst b/elements/components/linux.bst
index a9e7899f4..3699ee62e 100644
--- a/elements/components/linux.bst
+++ b/elements/components/linux.bst
@@ -244,7 +244,6 @@ config:
 
     enable BLK_MQ_VIRTIO
     module VIRTIO_CONSOLE
-    enable HW_RANDOM_VIRTIO
     module VIRTIO_MMIO
     module CRYPTO_DEV_VIRTIO
 
@@ -1102,6 +1101,20 @@ config:
         ;;
     esac
 
+
+    # Enable HW Random
+    enable HW_RANDOM
+    enable HW_RANDOM_VIRTIO
+    case "%{arch}" in
+      x86_64|i686)
+        enable HW_RANDOM_AMD
+        enable HW_RANDOM_INTEL
+        ;;
+      aarch64|arm)
+        enable HW_RANDOM_ARM_SMCCC_TRNG
+        ;;
+    esac
+
     # NVMEM support
     # Needed for USB4 and Thunderbolt
     enable MTD
-- 
2.43.0


From 5c2a81749a9ceb0843f6078fdbeec972a4521d64 Mon Sep 17 00:00:00 2001
From: Jordan Petridis <jordan@centricular.com>
Date: Fri, 8 Dec 2023 13:32:03 +0200
Subject: [PATCH 08/21] components/linux.bst: Enable ADRESS_MASKING

---
 elements/components/linux.bst | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/elements/components/linux.bst b/elements/components/linux.bst
index 3699ee62e..857e285c1 100644
--- a/elements/components/linux.bst
+++ b/elements/components/linux.bst
@@ -1101,6 +1101,15 @@ config:
         ;;
     esac
 
+    if has ARCH_SUPPORTS_MEMORY_FAILURE; then
+      enable MEMORY_FAILURE
+    fi
+
+    case "%{arch}" in
+      x86_64)
+        enable ADDRESS_MASKING
+        ;;
+    esac
 
     # Enable HW Random
     enable HW_RANDOM
-- 
2.43.0


From 53691dfbd87474dc777fe65cf824a767b1f0a715 Mon Sep 17 00:00:00 2001
From: Jordan Petridis <jordan@centricular.com>
Date: Fri, 8 Dec 2023 13:37:10 +0200
Subject: [PATCH 09/21] components/linux.bst: Enable more kvm features

---
 elements/components/linux.bst | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/elements/components/linux.bst b/elements/components/linux.bst
index 857e285c1..ea6068ec8 100644
--- a/elements/components/linux.bst
+++ b/elements/components/linux.bst
@@ -1291,9 +1291,11 @@ config:
     # KVM
     case "%{arch}" in
       x86_64|i686)
+        enable PARAVIRT_SPINLOCKS
+        enable PARAVIRT_TIME_ACCOUNTING
         module KVM
-        module KVM_INTEL
         module KVM_AMD
+        module KVM_INTEL
         ;;
       aarch64)
         module KVM
-- 
2.43.0


From 2225c0c9aeb7c9b4f51f9beec542ab3fb2cad2b8 Mon Sep 17 00:00:00 2001
From: Jordan Petridis <jordan@centricular.com>
Date: Fri, 8 Dec 2023 14:05:47 +0200
Subject: [PATCH 10/21] components/linux.bst: Enable more acpi and cpufreq
 modules

---
 elements/components/linux.bst | 76 ++++++++++++++++++++++++++++++++++-
 1 file changed, 74 insertions(+), 2 deletions(-)

diff --git a/elements/components/linux.bst b/elements/components/linux.bst
index ea6068ec8..8b28851de 100644
--- a/elements/components/linux.bst
+++ b/elements/components/linux.bst
@@ -1081,6 +1081,66 @@ config:
       ;;
     esac
 
+    # ACPI
+    enable ACPI_FFH
+    enable ACPI_PCI_SLOT
+    module ACPI_HED
+    module ACPI_IPMI
+    module ACPI_PFRUT
+    module ACPI_TAD
+
+    if has ARCH_ENABLE_MEMORY_HOTPLUG; then
+      enable MEMORY_HOTPLUG
+      enable ACPI_HOTPLUG_MEMORY
+    fi
+
+    if has ACPI_NUMA; then
+      enable ACPI_HMAT
+    fi
+
+    case "%{arch}" in
+      x86_64|aarch64)
+        enable ACPI_FPDT
+        ;;
+    esac
+
+    # acpi apei
+    if has HAVE_ACPI_APEI; then
+      enable ACPI_APEI
+      enable ACPI_APEI_GHES
+      if has ARCH_SUPPORTS_MEMORY_FAILURE; then
+        enable ACPI_APEI_MEMORY_FAILURE
+      fi
+      enable ACPI_APEI_PCIEAER
+      module ACPI_APEI_EINJ
+
+      case "%{arch}" in
+        aarch64)
+          enable ACPI_APEI_SEA
+          ;;
+      esac
+    fi
+
+    case "%{arch}" in
+      x86_64|i686)
+        module ACPI_EXTLOG
+        module ACPI_PROCESSOR_AGGREGATOR
+        module ACPI_SBS
+        ;;
+    esac
+
+    # acpi pmic
+    case "%{arch}" in
+      x86_64|i686)
+        enable BXT_WC_PMIC_OPREGION
+        enable BYTCRC_PMIC_OPREGION
+        enable CHTCRC_PMIC_OPREGION
+        enable CHT_DC_TI_PMIC_OPREGION
+        enable CHT_WC_PMIC_OPREGION
+        enable XPOWER_PMIC_OPREGION
+        ;;
+    esac
+
     # HID drivers
     module HID_MULTITOUCH
     module HID_GENERIC
@@ -1377,6 +1437,10 @@ config:
       module EDAC
     fi
 
+    if has HAS_IOMEM; then
+      module IPMI_HANDLER
+    fi
+
     # PCI
     if has HAVE_PCI; then
       enable PCI_IOV
@@ -1840,15 +1904,23 @@ config:
     # cpufreq
     case "%{arch}" in
       x86_64|i686)
+        enable CPU_FREQ_STAT
+        enable X86_AMD_PSTATE
         module CPU_FREQ_GOV_CONSERVATIVE
+        module CPU_FREQ_GOV_PERFORMANCE
         module CPU_FREQ_GOV_POWERSAVE
-        enable CPU_FREQ_STAT
         module X86_AMD_FREQ_SENSITIVITY
-        enable X86_AMD_PSTATE
+        module X86_AMD_PSTATE_UT
         module X86_P4_CLOCKMOD
         module X86_PCC_CPUFREQ
         module X86_POWERNOW_K8
       ;;
+      aarch64|arm)
+        enable ACPI_CPPC_CPUFREQ_FIE
+        enable ARM_PSCI_CPUIDLE_DOMAIN
+        module ACPI_CPPC_CPUFREQ
+        module ARM_SCPI_CPUFREQ
+      ;;
     esac
 
     # Platforms
-- 
2.43.0


From f73e36d4f8182c60ada45e09eb6075b55c57b1af Mon Sep 17 00:00:00 2001
From: Jordan Petridis <jordan@centricular.com>
Date: Fri, 8 Dec 2023 15:01:47 +0200
Subject: [PATCH 11/21] components/linux.bst: Enable more platforms

---
 elements/components/linux.bst | 67 +++++++++++++++++++++++++++++++++--
 1 file changed, 64 insertions(+), 3 deletions(-)

diff --git a/elements/components/linux.bst b/elements/components/linux.bst
index 8b28851de..061ec212a 100644
--- a/elements/components/linux.bst
+++ b/elements/components/linux.bst
@@ -280,6 +280,7 @@ config:
     module TOUCHSCREEN_USB_COMPOSITE
     enable TOUCHSCREEN_USB_E2I
     module TOUCHSCREEN_WACOM_W8001
+    module TOUCHSCREEN_SILEAD
 
     # tablets
     module TABLET_SERIAL_WACOM4
@@ -335,10 +336,21 @@ config:
     # Needed by some devices
     case "%{arch}" in
       i686|x86_64)
+        enable INTEL_TPMI
+        enable INTEL_VSEC
         enable X86_INTEL_LPSS
       ;;
     esac
 
+    # Device Tree and Open Firmware
+    case "%{arch}" in
+      aarch64|arm)
+        enable DTPM
+        enable DTPM_CPU
+        enable DTPM_DEVFREQ
+      ;;
+    esac
+
     # Wireguard
     module WIREGUARD
 
@@ -1454,6 +1466,8 @@ config:
       if has RAS; then
         enable PCIEAER
         module PCIEAER_INJECT
+        enable PCIEASPM
+        enable PCIEASPM_POWER_SUPERSAVE
         enable PCIE_ECRC
         enable PCIE_DPC
         if has ACPI; then
@@ -1925,12 +1939,38 @@ config:
 
     # Platforms
     case "%{arch}" in
-      x86_64|i686)
-        module CHROMEOS_LAPTOP
-        module CHROMEOS_PSTORE
+      aarch64|arm|x86_64|i686)
+        # Chrome Platform
         enable CHROME_PLATFORMS
+        module CHARGER_CROS_USBPD
+        module CHROMEOS_ACPI
+        module CHROMEOS_PRIVACY_SCREEN
+        module CHROMEOS_TBMC
         module CROS_EC
+        module CROS_EC_I2C
+        module CROS_EC_SPI
+        module CROS_EC_UART
+        module CROS_EC_UART
+        module CROS_HPS_I2C
+        module CROS_HPS_I2C
         module CROS_KBD_LED_BACKLIGHT
+        module CROS_USBPD_LOGGER
+
+        # Mellanox Platform
+        enable MELLANOX_PLATFORM
+        module MLXREG_HOTPLUG
+        module MLXREG_IO
+        module MLXREG_LC
+        module NVSW_SN2201
+      ;;
+    esac
+
+    case "%{arch}" in
+      x86_64|i686)
+        enable MLX_PLATFORM
+        module CHROMEOS_LAPTOP
+        module CHROMEOS_PSTORE
+        module CROS_EC_LPC
         module SURFACE_AGGREGATOR
         module SURFACE_AGGREGATOR_CDEV
         module SURFACE_AGGREGATOR_HUB
@@ -1947,6 +1987,7 @@ config:
         module ACER_WIRELESS
         module ACPI_TOSHIBA
         module AMD_PMC
+        module AMD_PMF
         module AMILO_RFKILL
         module APPLE_GMUX
         module ASUS_LAPTOP
@@ -1969,30 +2010,38 @@ config:
         module INTEL_IPS
         module INTEL_PMC_CORE
         module LG_LAPTOP
+        module LENOVO_YMC
         module MSI_LAPTOP
+        module MSI_EC
         module PANASONIC_LAPTOP
         module PCENGINES_APU2
         module SAMSUNG_LAPTOP
         module SAMSUNG_Q10
         module SENSORS_HDAPS
         module SERIAL_MULTI_INSTANTIATE
+        module SYSTEM76_ACPI
         module SONY_LAPTOP
         enable SONYPI_COMPAT
         module THINKPAD_ACPI
         enable THINKPAD_ACPI_ALSA_SUPPORT
         enable THINKPAD_ACPI_HOTKEY_POLL
         enable THINKPAD_ACPI_VIDEO
+        enable THINKPAD_LMI
         module TOPSTAR_LAPTOP
         module TOSHIBA_BT_RFKILL
         module TOSHIBA_HAPS
         module TOSHIBA_WMI
+        module WIRELESS_HOTKEY
+        module YOGABOOK
         module ACPI_WMI
         module ACER_WMI
+        module GIGABYTE_WMI
         enable X86_PLATFORM_DRIVERS_DELL
         enable X86_PLATFORM_DRIVERS_HP
         module ALIENWARE_WMI
         module ASUS_NB_WMI
         module ASUS_WMI
+        module ASUS_TF103C_DOCK
         enable DELL_SMBIOS_WMI
         module DELL_WMI_AIO
         module DELL_WMI_LED
@@ -2009,6 +2058,18 @@ config:
       ;;
     esac
 
+    # power
+    enable ENERGY_MODEL
+
+    # Powercap
+    enable POWERCAP
+    enable IDLE_INJECT
+    case "%{arch}" in
+      x86_64|i686)
+        module INTEL_RAPL
+      ;;
+    esac
+
   - |
     make -j1 olddefconfig
 
-- 
2.43.0


From 07540cb567c6186450ed70e4a94c10e3ca1e608c Mon Sep 17 00:00:00 2001
From: Jordan Petridis <jordan@centricular.com>
Date: Fri, 8 Dec 2023 16:26:40 +0200
Subject: [PATCH 12/21] components/linux.bst: Enable INTEL_IOMMU

---
 elements/components/linux.bst | 1 +
 1 file changed, 1 insertion(+)

diff --git a/elements/components/linux.bst b/elements/components/linux.bst
index 061ec212a..3d1d0d241 100644
--- a/elements/components/linux.bst
+++ b/elements/components/linux.bst
@@ -1169,6 +1169,7 @@ config:
         # pinctrl_amd might not work as module if it is loaded to late
         enable PINCTRL_AMD
         enable AMD_IOMMU_V2
+        enable INTEL_IOMMU
         enable X86_AMD_PLATFORM_DEVICE
         ;;
     esac
-- 
2.43.0


From 6ece420b15c7838f2e9b6591158d35d9a882a563 Mon Sep 17 00:00:00 2001
From: Jordan Petridis <jordan@centricular.com>
Date: Fri, 8 Dec 2023 18:36:23 +0200
Subject: [PATCH 13/21] components/linux.bst: Enable more filesystem features

---
 elements/components/linux.bst | 24 +++++++++++++++++++++---
 1 file changed, 21 insertions(+), 3 deletions(-)

diff --git a/elements/components/linux.bst b/elements/components/linux.bst
index 3d1d0d241..3de0b431b 100644
--- a/elements/components/linux.bst
+++ b/elements/components/linux.bst
@@ -220,9 +220,14 @@ config:
     module OVERLAY_FS
 
     # Required by snapd
-    enable SQUASHFS_XZ
+    enable SQUASHFS_COMPILE_DECOMP_MULTI_PERCPU
+    enable SQUASHFS_DECOMP_MULTI_PERCPU
+    enable SQUASHFS_FILE_DIRECT
+    enable SQUASHFS_LZ4
     enable SQUASHFS_LZO
     enable SQUASHFS_XATTR
+    enable SQUASHFS_XZ
+    enable SQUASHFS_ZSTD
 
     # erofs is potential replacement of squashfs on GNOME OS
     module EROFS_FS
@@ -231,6 +236,7 @@ config:
     enable EROFS_FS_SECURITY
     enable EROFS_FS_ZIP
     enable EROFS_FS_ZIP_LZMA
+    enable EROFS_FS_ZIP_DEFLATE
 
     # Some useful drivers when running as virtual machine
     module VIRTIO_BALLOON
@@ -659,8 +665,9 @@ config:
     fi
 
     # FUSE
-    module FUSE_FS
     module CUSE
+    module FUSE_FS
+    module VIRTIO_FS
 
     # iSCSI
     enable SCSI_LOWLEVEL
@@ -1380,16 +1387,27 @@ config:
     for fs in XFS BTRFS EXFAT VFAT F2FS ISO9660 UDF NTFS3; do
       module "${fs}_FS"
     done
-    for fs in BTRFS F2FS; do
+    for fs in BTRFS F2FS NTFS3; do
       enable "${fs}_FS_POSIX_ACL"
     done
+    enable "NTFS3_LZX_XPRESS"
     enable "XFS_POSIX_ACL"
     enable FS_ENCRYPTION
+    enable XFS_ONLINE_SCRUB
+    enable XFS_QUOTA
+    enable XFS_RT
 
     # Needed for FAT filesystems
     enable NLS_CODEPAGE_437
     enable NLS_ISO8859_1
 
+    # Will be needed for composefs
+    enable FS_VERITY
+
+    # FS access notifications
+    enable FANOTIFY
+    enable FANOTIFY_ACCESS_PERMISSIONS
+
     # Extcon
     module EXTCON
     case "%{arch}" in
-- 
2.43.0


From 1fa54b12e55c1bb04d338a8dc9d5c87983f1bad7 Mon Sep 17 00:00:00 2001
From: Jordan Petridis <jordan@centricular.com>
Date: Fri, 8 Dec 2023 19:40:26 +0200
Subject: [PATCH 14/21] components/linux.bst: Enable power regulators

---
 elements/components/linux.bst | 60 +++++++++++++++++++++++++++++++++++
 1 file changed, 60 insertions(+)

diff --git a/elements/components/linux.bst b/elements/components/linux.bst
index 3de0b431b..3488ebb5a 100644
--- a/elements/components/linux.bst
+++ b/elements/components/linux.bst
@@ -1431,6 +1431,66 @@ config:
       ;;
     esac
 
+    # Regulators
+    enable REGULATOR
+    module REGULATOR_FIXED_VOLTAGE
+    module REGULATOR_VIRTUAL_CONSUMER
+    module REGULATOR_USERSPACE_CONSUMER
+    module REGULATOR_88PG86X
+    module REGULATOR_ACT8865
+    module REGULATOR_AD5398
+    module REGULATOR_AW37503
+    module REGULATOR_AXP20X
+    module REGULATOR_DA9210
+    module REGULATOR_DA9211
+    module REGULATOR_FAN53555
+    module REGULATOR_GPIO
+    module REGULATOR_ISL9305
+    module REGULATOR_ISL6271A
+    module REGULATOR_LP3971
+    module REGULATOR_LP3972
+    module REGULATOR_LP872X
+    module REGULATOR_LP8755
+    module REGULATOR_LTC3589
+    module REGULATOR_LTC3676
+    module REGULATOR_MAX1586
+    module REGULATOR_MAX77857
+    module REGULATOR_MAX8649
+    module REGULATOR_MAX8660
+    module REGULATOR_MAX8893
+    module REGULATOR_MAX8952
+    module REGULATOR_MAX20086
+    module REGULATOR_MAX20411
+    module REGULATOR_MAX77826
+    module REGULATOR_MP8859
+    module REGULATOR_MT6311
+    module REGULATOR_PCA9450
+    module REGULATOR_PV88060
+    module REGULATOR_PV88080
+    module REGULATOR_PV88090
+    module REGULATOR_PWM
+    module REGULATOR_RAA215300
+    module REGULATOR_RT4801
+    module REGULATOR_RT4803
+    module REGULATOR_RT5190A
+    module REGULATOR_RT5739
+    module REGULATOR_RT5759
+    module REGULATOR_RT6160
+    module REGULATOR_RT6190
+    module REGULATOR_RT6245
+    module REGULATOR_RTQ2134
+    module REGULATOR_RTMV20
+    module REGULATOR_RTQ6752
+    module REGULATOR_RTQ2208
+    module REGULATOR_SLG51000
+    module REGULATOR_TPS51632
+    module REGULATOR_TPS62360
+    module REGULATOR_TPS65023
+    module REGULATOR_TPS6507X
+    module REGULATOR_TPS65132
+    module REGULATOR_TPS6524X
+    module RC_CORE
+
     # PWM
     enable PWM
     case "%{arch}" in
-- 
2.43.0


From 2edeab2fc1b0e40129697cd8ecf651bf51180d51 Mon Sep 17 00:00:00 2001
From: Jordan Petridis <jordan@centricular.com>
Date: Fri, 8 Dec 2023 20:10:06 +0200
Subject: [PATCH 15/21] components/linux.bst: Enable ZSWAP

---
 elements/components/linux.bst | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/elements/components/linux.bst b/elements/components/linux.bst
index 3488ebb5a..4829a797d 100644
--- a/elements/components/linux.bst
+++ b/elements/components/linux.bst
@@ -1917,6 +1917,9 @@ config:
       enable BPF_JIT
     fi
 
+    # ZSWAP
+    enable ZSWAP
+
     # Secure boot
     enable SYSTEM_BLACKLIST_KEYRING
     enable SYSTEM_TRUSTED_KEYRING
-- 
2.43.0


From 47ef0b77c7362bcac53c15ee2a8c0768e45de32d Mon Sep 17 00:00:00 2001
From: Jordan Petridis <jordan@centricular.com>
Date: Fri, 8 Dec 2023 20:50:15 +0200
Subject: [PATCH 16/21] components/linux.bst: Enable power related modules

---
 elements/components/linux.bst | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/elements/components/linux.bst b/elements/components/linux.bst
index 4829a797d..7230189f5 100644
--- a/elements/components/linux.bst
+++ b/elements/components/linux.bst
@@ -2142,6 +2142,25 @@ config:
 
     # power
     enable ENERGY_MODEL
+    enable POWER_RESET
+    enable POWER_RESET_RESTART
+    enable POWER_SUPPLY
+    enable POWER_SUPPLY_HWMON
+
+    # power/supply
+    module BATTERY_CW2015
+    module BATTERY_RT5033
+    module BATTERY_UG3105
+    module CHARGER_BD99954
+    module CHARGER_BQ2515X
+    module CHARGER_BQ256XX
+    module CHARGER_BQ25890
+    module CHARGER_LT3651
+    module CHARGER_LTC4162L
+    module CHARGER_MAX77976
+    module CHARGER_RT9467
+    module CHARGER_RT9471
+    module CHARGER_SMB347
 
     # Powercap
     enable POWERCAP
-- 
2.43.0


From e4fe7f7ca956edf27f08829d161b9de70483f6d9 Mon Sep 17 00:00:00 2001
From: Jordan Petridis <jordan@centricular.com>
Date: Thu, 7 Dec 2023 21:26:36 +0200
Subject: [PATCH 17/21] components/linux.bst: Use UNWINDER_FRAME_POINTER
 instead of UNWINDER_ORC

---
 elements/components/linux.bst | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/elements/components/linux.bst b/elements/components/linux.bst
index 7230189f5..4980ba30d 100644
--- a/elements/components/linux.bst
+++ b/elements/components/linux.bst
@@ -156,6 +156,17 @@ config:
     # Userspace firmware loading not supported
     remove FW_LOADER_USER_HELPER
 
+    # UNWINDER_ORC is great for the kernel
+    # but it wasn't designed to be used from the userspace
+    # thus we need FRAME_POINTER so userspace applications
+    # will also be able to capture traces efficiently
+    case "%{arch}" in
+      x86_64|i686)
+        remove UNWINDER_ORC
+        enable UNWINDER_FRAME_POINTER
+      ;;
+    esac
+
     # Some udev/virtualization requires
     enable DMIID
 
-- 
2.43.0


From 6f31704a0da2b37565b7bddc07c162f4f92237b0 Mon Sep 17 00:00:00 2001
From: Jordan Petridis <jordan@centricular.com>
Date: Sat, 9 Dec 2023 17:52:07 +0200
Subject: [PATCH 18/21] components/linux.bst: Enable IOMMUFD and IRQ_REMAP

---
 elements/components/linux.bst | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/elements/components/linux.bst b/elements/components/linux.bst
index 4980ba30d..02f4d0c56 100644
--- a/elements/components/linux.bst
+++ b/elements/components/linux.bst
@@ -1182,12 +1182,16 @@ config:
 
     # CPU support
     enable PINCTRL
+    if has IOMMU_SUPPORT; then
+      enable IOMMUFD
+    fi
     case "%{arch}" in
       x86_64|i686)
         # pinctrl_amd might not work as module if it is loaded to late
         enable PINCTRL_AMD
         enable AMD_IOMMU_V2
         enable INTEL_IOMMU
+        enable IRQ_REMAP
         enable X86_AMD_PLATFORM_DEVICE
         ;;
     esac
-- 
2.43.0


From ab9e2fea2d3e215f82d9ec45d05a372ff8cd8e6c Mon Sep 17 00:00:00 2001
From: Jordan Petridis <jordan@centricular.com>
Date: Sat, 9 Dec 2023 17:57:03 +0200
Subject: [PATCH 19/21] components/linux.bst: Explicitly enable PERF_EVENTS

Currently only the _INTEL and _AMD_UNCORE are enabled in the default
config, and since we are here, explicitly enable all of them so they
won't go missing without us noticing.
---
 elements/components/linux.bst | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/elements/components/linux.bst b/elements/components/linux.bst
index 02f4d0c56..cd747262b 100644
--- a/elements/components/linux.bst
+++ b/elements/components/linux.bst
@@ -1206,6 +1206,18 @@ config:
         ;;
     esac
 
+    # Performance monitoring
+    case "%{arch}" in
+      x86_64|i686)
+        enable PERF_EVENTS_AMD_BRS
+        enable PERF_EVENTS_INTEL_CSTATE
+        enable PERF_EVENTS_INTEL_RAPL
+        enable PERF_EVENTS_INTEL_UNCORE
+        module PERF_EVENTS_AMD_POWER
+        module PERF_EVENTS_AMD_UNCORE
+      ;;
+    esac
+
     # Enable HW Random
     enable HW_RANDOM
     enable HW_RANDOM_VIRTIO
-- 
2.43.0


From c3fc4d1ade83c0873979329fcdfc08e4056054c6 Mon Sep 17 00:00:00 2001
From: Jordan Petridis <jordan@centricular.com>
Date: Sat, 9 Dec 2023 18:24:18 +0200
Subject: [PATCH 20/21] components/linux.bst: enable shadow stack support when
 possible

---
 elements/components/linux.bst | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/elements/components/linux.bst b/elements/components/linux.bst
index cd747262b..71af3346f 100644
--- a/elements/components/linux.bst
+++ b/elements/components/linux.bst
@@ -1193,6 +1193,7 @@ config:
         enable INTEL_IOMMU
         enable IRQ_REMAP
         enable X86_AMD_PLATFORM_DEVICE
+        enable CONFIG_X86_USER_SHADOW_STACK
         ;;
     esac
 
@@ -1200,6 +1201,10 @@ config:
       enable MEMORY_FAILURE
     fi
 
+    if has ARCH_SUPPORTS_SHADOW_CALL_STACK; then
+      enable SHADOW_CALL_STACK
+    fi
+
     case "%{arch}" in
       x86_64)
         enable ADDRESS_MASKING
-- 
2.43.0


From f8fa31e829de4da255386dd74fc7f42c2f324088 Mon Sep 17 00:00:00 2001
From: Jordan Petridis <jordan@centricular.com>
Date: Sat, 9 Dec 2023 19:09:55 +0200
Subject: [PATCH 21/21] components/linux.bst: Enable IRQ_TIME_ACCOUNTING

---
 elements/components/linux.bst | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/elements/components/linux.bst b/elements/components/linux.bst
index 71af3346f..1111967e5 100644
--- a/elements/components/linux.bst
+++ b/elements/components/linux.bst
@@ -1205,6 +1205,10 @@ config:
       enable SHADOW_CALL_STACK
     fi
 
+    if has HAVE_IRQ_TIME_ACCOUNTING; then
+      enable IRQ_TIME_ACCOUNTING
+    fi
+
     case "%{arch}" in
       x86_64)
         enable ADDRESS_MASKING
-- 
2.43.0

