Only in freedesktop-sdk.old/: .bstproject.yaml
diff -ur freedesktop-sdk.old/files/linux-pam-config/smartcard-auth freedesktop-sdk/files/linux-pam-config/smartcard-auth
--- freedesktop-sdk.old/files/linux-pam-config/smartcard-auth	2025-10-04 16:19:43.114390723 +0200
+++ freedesktop-sdk/files/linux-pam-config/smartcard-auth	2025-10-04 16:24:03.076738328 +0200
@@ -1,16 +1 @@
-auth        required      pam_env.so
-auth        [success=done ignore=ignore default=die] pam_pkcs11.so wait_for_card
-auth        required      pam_deny.so
-
-account     required      pam_unix.so
-account     sufficient    pam_localuser.so
-account     sufficient    pam_succeed_if.so uid < 500 quiet
-account     required      pam_permit.so
-
-password    optional      pam_pkcs11.so
-
-session     optional      pam_keyinit.so revoke
-session     required      pam_limits.so
--session     optional      pam_systemd.so
-session     [success=1 default=ignore] pam_succeed_if.so service in crond quiet use_uid
-session     required      pam_unix.so
+auth required pam_debug.so auth=authinfo_unavail
