diff --git a/elements/components/noopenh264.bst b/elements/components/noopenh264.bst
index 4e73f22..1f9f294 100644
--- a/elements/components/noopenh264.bst
+++ b/elements/components/noopenh264.bst
@@ -15,7 +15,8 @@ variables:
 config:
   configure-commands:
     (<):
-    - grep "matching_version = '%{openh264-version}'" %{build-root}/meson.build
+    - |
+      grep "version : '%{openh264-version}'" %{build-root}/meson.build
 
   install-commands:
     (>):
@@ -39,4 +40,5 @@ public:
 sources:
 - kind: git_repo
   url: fdsdk_git:noopenh264.git
-  ref: v2.4.1-2-0-g96d0e58260580ee6fcab4331bfc792309e3c6128
+  track: v*
+  ref: v2.5.0-0-g6e575e43922a7620165637b2596c9686aad1f923
diff --git a/include/repo_branches.yml b/include/repo_branches.yml
index c770711..ee372a1 100644
--- a/include/repo_branches.yml
+++ b/include/repo_branches.yml
@@ -4,4 +4,4 @@
 freedesktop-sdk-flatpak-branch: '24.08'
 freedesktop-sdk-flatpak-branch-extra: '24.08extra'
 freedesktop-sdk-snap-branch: '2408'
-openh264-version: '2.4.1'
+openh264-version: '2.5.0'
