diff --git a/elements/components/linux.bst b/elements/components/linux.bst
index 231e1330d..dbbb3eee9 100644
--- a/elements/components/linux.bst
+++ b/elements/components/linux.bst
@@ -1728,7 +1728,7 @@ config:
     module LEDS_TRIGGER_NETDEV
     module LEDS_TRIGGER_ONESHOT
     enable LEDS_TRIGGER_PANIC
-    module LEDS_TRIGGER_PATTERN
+    enable LEDS_TRIGGER_PATTERN
     module LEDS_TRIGGER_TIMER
     module LEDS_TRIGGER_TRANSIENT
     case "%{arch}" in
@@ -1836,6 +1836,14 @@ config:
         module SURFACE3_WMI
         module X86_ANDROID_TABLETS
       ;;
+      aarch64)
+        enable DRM_PANEL_SAMSUNG_SOFEF00
+        module QCOM_GSBI
+        enable SCHED_CLUSTER
+        module CHARGER_QCOM_SMB2
+        module QCOM_SPMI_RRADC
+        enable FB_SIMPLE
+      ;;
     esac
 
   - |
