commit bd842e69ad67a19e9233b22ffe860ab2d881b0ad
Author: Valentin David <me@valentindavid.com>
Date:   Thu Jun 19 19:34:54 2025 +0200

    components/sof-firmware.bst: Update to 2025.05

diff --git a/elements/components/sof-firmware.bst b/elements/components/sof-firmware.bst
index 10abb384e..9b069f6e0 100644
--- a/elements/components/sof-firmware.bst
+++ b/elements/components/sof-firmware.bst
@@ -24,5 +24,5 @@ config:
 
 sources:
 - kind: tar
-  url: github_files:thesofproject/sof-bin/releases/download/v2025.01/sof-bin-2025.01.tar.gz
-  ref: 34d565db757a32450106317cc51f38bf67962e0fc8b7f7c72e6e39fd89e99263
+  url: github_files:thesofproject/sof-bin/releases/download/v2025.05/sof-bin-2025.05.tar.gz
+  ref: e2f2603b0d38c7cbdb1672901863fbf84b4db4921f405952a236915cb8a86bcc
