diff --git a/files/linux/fdsdk-config.sh b/files/linux/fdsdk-config.sh
index 740178735..d514f583a 100644
--- a/files/linux/fdsdk-config.sh
+++ b/files/linux/fdsdk-config.sh
@@ -1593,6 +1593,11 @@ case "$arch" in
     ;;
 esac
 module VHOST_VSOCK
+module VHOST_NET
+module TARGET_CORE
+module VHOST_SCSI
+module VDPA_USER
+module VHOST_VDPA
 
 # Useful filesystems
 for fs in XFS BTRFS EXFAT VFAT F2FS ISO9660 UDF NTFS3; do
