diff --git a/files/linux/fdsdk-config.sh b/files/linux/fdsdk-config.sh
index ad581e6da..522d50e09 100644
--- a/files/linux/fdsdk-config.sh
+++ b/files/linux/fdsdk-config.sh
@@ -7,6 +7,51 @@ arch=$1
 
 # Modify the kernel config for additional features
 
+case "$arch" in
+    aarch64)
+        module VIDEO_ROCKCHIP_ISP1
+        module VIDEO_ROCKCHIP_RGA
+        module VIDEO_ROCKCHIP_VDEC
+        module VIDEO_HANTRO
+        module VIDEO_OV8858
+        module VIDEO_IMX258
+        module VIDEO_DW9714
+    ;;
+esac
+
+case "$arch" in
+    aarch64)
+        module ROCKCHIP_SARADC
+        enable ROCKCHIP_MBOX
+        module ROCKCHIP_THERMAL
+        enable ROCKCHIP_RGB
+    ;;
+esac
+
+case "$arch" in
+    aarch64)
+        module CRYPTO_DEV_ROCKCHIP
+    ;;
+esac
+
+case "$arch" in
+    aarch64)
+        module PHY_ROCKCHIP_DPHY_RX0
+    ;;
+esac
+
+case "$arch" in
+    aarch64)
+        enable BACKLIGHT_CLASS_DEVICE
+    ;;
+esac
+
+case "$arch" in
+    aarch64)
+        module V4L2_FLASH_LED_CLASS
+    ;;
+esac
+
 # Kernel Config Options
 enable DEVTMPFS
 enable CGROUPS
@@ -226,6 +271,10 @@ case "$arch" in
     i686|x86_64)
         module KEYBOARD_APPLESPI
     ;;
+    aarch64)
+        module INPUT_GPIO_VIBRA
+        module KEYBOARD_PINEPHONE
+    ;;
 esac
 module KEYBOARD_GPIO
 
@@ -619,6 +668,7 @@ case "$arch" in
     ;;
     aarch64)
         module DRM_PANFROST
+        module DRM_PANEL_HIMAX_HX8394
     ;;
 esac
 
@@ -2031,6 +2081,10 @@ case "$arch" in
         module LEDS_APU
         module LEDS_INTEL_SS4200
     ;;
+    aarch64)
+        module LEDS_CLASS_FLASH
+        module LEDS_SGM3140
+    ;;
 esac
 
 # cpufreq
