diff --git a/Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMEventTarget.h b/Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMEventTarget.h
index 75aac1894386..f060003e0baa 100644
--- a/Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMEventTarget.h
+++ b/Source/WebKit/WebProcess/InjectedBundle/API/gtk/DOM/WebKitDOMEventTarget.h
@@ -92,7 +92,7 @@ WEBKIT_DEPRECATED gboolean  webkit_dom_event_target_add_event_listener(WebKitDOM
  * webkit_dom_event_target_remove_event_listener:
  * @target: A #WebKitDOMEventTarget
  * @event_name: A #gchar
- * @handler: (scope call): A #GCallback
+ * @handler: (type gpointer): A #GCallback
  * @use_capture: A #gboolean
  *
  * Returns: a #gboolean
